pyaerocom/aeroval/experiment_output.py

```diff
@@ -766,9 +841,75 @@ def _is_part_of_experiment(self, obs_name, obs_var, mod_name, mod_var) -> bool:
-            (obs_name, obs_var, vert_code, mod_name, mod_var, per) = self._info_from_map_file(file)
+            if self.cfg.processing_opts.only_model_maps:
+                # Hack to build menu.json
+                # Key issue we need to get around is that the ExperimentOutput class
+                # expects that at this point in the processing, all information it needs
+                # to describe an experiment has been written to disc, traditionally in the map directory
+                # if only_model_maps = True, then we do not do colocation, and so the maps dir is empty,
+                # however menu.json is still needed
+                if not all_combinations:
+                    break
+
+                (mod_name, var_name, per) = self._info_from_contour_dir_file(file)
+
+                obs_var, mod_var = var_name, var_name
+
+                if mod_name in self.cfg.obs_cfg.keylist():
+                    obs_name = mod_name
+                    vert_code = self.cfg.obs_cfg.get_entry(obs_name).obs_vert_type
+                    first_with_obs_name = next(
+                        (
+                            item
+                            for item in all_combinations
+                            if item[0] == obs_name and item[-1] == var_name
+                        ),
+                        None,
+                    )
+                    if not first_with_obs_name:  # should already be taken care of in new
+                        continue
+                    mod_name = first_with_obs_name[1]
+                    all_combinations.remove(first_with_obs_name)
+                elif mod_name in self.cfg.model_cfg.keylist():
+                    vert_code = None
+                    for o in self.cfg.obs_cfg.keylist():
+                        if var_name in self.cfg.obs_cfg.get_entry(o).obs_vars:
+                            vert_code = self.cfg.obs_cfg.get_entry(o).obs_vert_type
+                    if not vert_code:
+                        raise ValueError(
+                            "Failed to infer vert_code in a only_model_maps experiment"
+                        )
+                    first_with_mod_name = next(
+                        (
+                            item
+                            for item in all_combinations
+                            if item[1] == mod_name and item[-1] == var_name
+                        ),
+                        None,
+                    )
+                    if not first_with_mod_name:  # should already be taken care of in new
+                        continue
+                    obs_name = first_with_mod_name[0]
+                    all_combinations.remove(first_with_mod_name)
+                else:
+                    raise ValueError("Failed to infer vert_code in a only_model_maps experiment")
+
+            else:
+                (obs_name, obs_var, vert_code, mod_name, mod_var, per) = self._info_from_map_file(
+                    file
+                )
```

Comment:
I agree that this looks very hacky, just to get `model_maps_only`. Since splitting the output-parts into smaller bits is very welcome from a point of performance, maybe the default of trying to do everything is wrong?

Reply:
It is very hacky and I wish it didn't need to be, but I just don't see a way around it with out much bigger changes in our processing of experiments. I agree, that splitting things into smaller parts is preferred. Definitely we should think about a better approach.